python/ray/data/datasource/mongo_datasource.py

```diff
@@ -0,0 +1,185 @@
+import logging
+from typing import Any, Dict, List, Optional
+
+from ray.data.datasource.datasource import Datasource, Reader, ReadTask, WriteResult
+from ray.data.block import (
+    Block,
+    BlockMetadata,
+)
+from ray.data._internal.remote_fn import cached_remote_fn
+from ray.types import ObjectRef
+from ray.util.annotations import PublicAPI
+
+import pymongo
+from pymongoarrow.api import Schema
+
+logger = logging.getLogger(__name__)
+
+
+@PublicAPI(stability="alpha")
```

Comment:
actually is this public API? User should never use this but `read_mongo` or `write_mongo` instead?